src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -446,11 +449,31 @@ public void handle_package_result(PackageResult packageResult, ChocolateyConfigu
             }
             else
             {
-                this.Log().Info(ChocolateyLoggers.Important, @"  Software install location not explicitly set, could be in package or 
+                this.Log().Info(ChocolateyLoggers.Important, @"  Software install location not explicitly set, could be in package or
   default install location if installer.");
             }
         }
 
+        private void create_ignore_files_for_executables(string installLocation, bool is64Bit)
+        {
+            // If we are using a 64 bit architecure, we want to ignore exe's targetting x86
+            // This is done by adding a .ignore file into the package folder for each exe to ignore
+            var exeFiles32Bit = _fileSystem.get_files(_fileSystem.combine_paths(installLocation, "tools\\x86"), pattern: "*.exe", option: SearchOption.AllDirectories).ToArray();
+            var exeFiles64Bit = _fileSystem.get_files(_fileSystem.combine_paths(installLocation, "tools\\x64"), pattern: "*.exe", option: SearchOption.AllDirectories).ToArray();
+
+            // If 64bit, and there are only 32bit files, we should shim the 32bit versions,
+            // therefore, don't ignore anything
+            if (!exeFiles64Bit.Any() && exeFiles32Bit.Any() && is64Bit)

```

Comment:
Micro-optimization: You should check `is64bit` first to short-circuit, and then you only need to check if there are no 64bit files prior to returning.

Reply:
Let me restate the second part of that. You only need to check if there are no 64bit files prior to returning when 64bit.